### PR TITLE
lxd/containers: Set liblxc env for CVE-2019-5736

### PR DESCRIPTION
**Pull request by stgraber**
Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>

### CODE REVIEW

**Comment by brauner** [comment]
Nice! Thanks!

**Comment by brauner** [comment]
Actually...
How about we only do this for privileged containers aka where we detect that an identity mapping for uid 0 is present?

**Comment by brauner** [comment]
Might be worth reporting such containers as privileged?

**Comment by stgraber** [comment]
@brauner updated, I think this should cover both privileged and root mapped unprivileged

File: lxd/container_lxc.go [hunk 1]
```diff
@@ -6068,6 +6068,23 @@ func (c *containerLXC) Exec(command []string, env map[string]string, stdin *os.F
 	cmd.Stdout = logFile
 	cmd.Stderr = logFile
 
+	// Mitigation for CVE-2019-5736
+	useRexec := false
+	if c.expandedConfig["raw.idmap"] != "" {
+		err := allowedUnprivilegedOnlyMap(c.expandedConfig["raw.idmap"])
+		if err != nil {
+			useRexec = true
+		}
```
Comment (brauner) [inline lxd/container_lxc.go]: So this still executes for all unprivileged containers not just ones where 0 is mapped to 0? I'm fine with that just checking.

Comment (stgraber) [reply lxd/container_lxc.go]: No, this should only set reExec if allowedUnprivilegedOnlyMap returns an error and it's only supposed to do that if root is mapped.

It's the same function that we use when LXD_UNPRIVILEGED_ONLY is set on Chromebooks.

Comment (brauner) [reply lxd/container_lxc.go]: Sweet.

**Pull request merged**